### DISCUSSION TITLE
Should renovate honour when repositories are disabled/archived in Azure repos?

### DISCUSSION THREAD

**Discussion by RolfMoleman** [renovatebot/renovate#29228] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Azure DevOps renovate version 37.374.3

### Please tell us more about your question or problem

We currently have an issue when running renovate via a centralised Azure DevOps pipeline where it throws multiple errors when trying to update a repository that has been set as disabled.

We are currently only targeting a subset of our projects and repositories using the autodiscoverFilter and have even tried to negate the unwanted repository using "!wantedproject/unwantedrepo" but this then caused renovate to target additional repositories outside of the proposed target set.

do we actually need to actively exclude a disabled repository or should renovate do this by default?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Renovate should ideally be able to detect and avoid this. any pointers on which fields to look in would be helpful

**Reply by RolfMoleman** [reply]
Apparel tly there is no clear field,  to specify if a repository is disabled in any api response. 

When trying to query a disabled repository using the azure DevOps repositories api it gives

```shell

{"$id":"1","innerException":null,"message":"TF401019: The Git repository with name or identifier repo_a does not exist or you do not have permissions for the operation you are attempting.","typeName":"Microsoft.TeamFoundation.Git.Server.GitRepositoryNotFoundException, Microsoft.TeamFoundation.Git.Server","typeKey":"GitRepositoryNotFoundException","errorCode":0,"eventId":3000}
```

I'm a project collection admin and knoe the repo in question exists but is disabled.

**Reply by rarkins** [reply]
So what would be the next best thing? Can renovate detect and skip it gracefully during the initRepo stage?

**Reply by RolfMoleman** [reply]
The workaround seems to be use regex to purposely ignore the repo in autodiscoverFilter. 

I'm going to do some more digging via the APIs to see if there is any reference to the repo being disabled and will report back. As so far it looks as if it just blocks access to all sees and gives a 403 is the api but a disabled page via the UI

**Reply by viceice** [reply]
https://github.com/renovatebot/renovate/blob/80232795f3352afeae67e6203ffae92e1515449f/lib/modules/platform/azure/index.ts#L114-L121

renovate should ignore disabled repos on autodiscover 🤔

**Reply by viceice** [reply]
https://github.com/renovatebot/renovate/blob/80232795f3352afeae67e6203ffae92e1515449f/lib/modules/platform/azure/index.ts#L205-L208

**Comment by RolfMoleman** [comment, marked as answer]
After further testing can confirm his works as described above. However, sometimes disabling the repository doesn't take effect, so simply re-enable and then disable again
